airbyte-workers/src/main/java/io/airbyte/workers/DefaultReplicationWorker.java

```diff
@@ -95,21 +106,59 @@ public StandardSyncOutput run(StandardSyncInput syncInput, Path jobRoot) throws
         destination.start(destinationConfig, jobRoot);
         source.start(sourceConfig, jobRoot);
 
-        while (!cancelled.get() && !source.isFinished()) {
-          final Optional<AirbyteMessage> maybeMessage = source.attemptRead();
-          if (maybeMessage.isPresent()) {
-            final AirbyteMessage message = mapper.mapMessage(maybeMessage.get());
-
-            messageTracker.accept(message);
-            destination.accept(message);
+        final Thread sourceThread = new Thread(() -> {
+          try {
+            while (!cancelled.get() && !source.isFinished()) {
+              final Optional<AirbyteMessage> messageOptional = source.attemptRead();
+              if (messageOptional.isPresent()) {
```

Comment:
there is a potential busy loop here. Shouldn't you have some sleep somewhere?

Reply:
confirmed it blocks while waiting for input.